### ISSUE TITLE
Package cannot be installed with pip

### ISSUE THREAD

**Issue by jbnjohnathan** [shellcheck-py/shellcheck-py#106]
Trying to install shellcheck-py==0.9.0.5 with python 3.10.4 on MacOS
```
Collecting shellcheck-py==0.9.0.5 (from -r requirements.txt (line 1))
  Using cached shellcheck_py-0.9.0.5.tar.gz (4.2 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting ruff==0.0.286 (from -r requirements.txt (line 2))
  Obtaining dependency information for ruff==0.0.286 from https://files.pythonhosted.org/packages/93/ba/8c6ba7d2314e231a761a596732a2b014538bd6c80142ffc759807721d64d/ruff-0.0.286-py3-none-macosx_10_9_x86_64.macosx_11_0_arm64.macosx_10_9_universal2.whl.metadata
  Downloading ruff-0.0.286-py3-none-macosx_10_9_x86_64.macosx_11_0_arm64.macosx_10_9_universal2.whl.metadata (21 kB)
Collecting black==23.7.0 (from -r requirements.txt (line 3))
  Obtaining dependency information for black==23.7.0 from https://files.pythonhosted.org/packages/10/bb/025dced0f7a2c00c59810700fbdab877b9a49cf817383133b79b0df5f0fe/black-23.7.0-cp310-cp310-macosx_10_16_universal2.whl.metadata
  Downloading black-23.7.0-cp310-cp310-macosx_10_16_universal2.whl.metadata (63 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 63.4/63.4 kB 6.1 MB/s eta 0:00:00
Collecting hadolintw==1.2.1 (from -r requirements.txt (line 4))
  Using cached hadolintw-1.2.1-py3-none-any.whl (5.2 kB)
Collecting click>=8.0.0 (from black==23.7.0->-r requirements.txt (line 3))
  Obtaining dependency information for click>=8.0.0 from https://files.pythonhosted.org/packages/00/2e/d53fa4befbf2cfa713304affc7ca780ce4fc1fd8710527771b58311a3229/click-8.1.7-py3-none-any.whl.metadata
  Downloading click-8.1.7-py3-none-any.whl.metadata (3.0 kB)
Collecting mypy-extensions>=0.4.3 (from black==23.7.0->-r requirements.txt (line 3))
  Using cached mypy_extensions-1.0.0-py3-none-any.whl (4.7 kB)
Collecting packaging>=22.0 (from black==23.7.0->-r requirements.txt (line 3))
  Using cached packaging-23.1-py3-none-any.whl (48 kB)
Collecting pathspec>=0.9.0 (from black==23.7.0->-r requirements.txt (line 3))
  Obtaining dependency information for pathspec>=0.9.0 from https://files.pythonhosted.org/packages/b4/2a/9b1be29146139ef459188f5e420a66e835dda921208db600b7037093891f/pathspec-0.11.2-py3-none-any.whl.metadata
  Downloading pathspec-0.11.2-py3-none-any.whl.metadata (19 kB)
Collecting platformdirs>=2 (from black==23.7.0->-r requirements.txt (line 3))
  Obtaining dependency information for platformdirs>=2 from https://files.pythonhosted.org/packages/14/51/fe5a0d6ea589f0d4a1b97824fb518962ad48b27cd346dcdfa2405187997a/platformdirs-3.10.0-py3-none-any.whl.metadata
  Downloading platformdirs-3.10.0-py3-none-any.whl.metadata (11 kB)
Requirement already satisfied: tomli>=1.1.0 in ./venv/lib/python3.10/site-packages (from black==23.7.0->-r requirements.txt (line 3)) (2.0.1)
Using cached ruff-0.0.286-py3-none-macosx_10_9_x86_64.macosx_11_0_arm64.macosx_10_9_universal2.whl (10.8 MB)
Using cached black-23.7.0-cp310-cp310-macosx_10_16_universal2.whl (2.7 MB)
Using cached click-8.1.7-py3-none-any.whl (97 kB)
Using cached pathspec-0.11.2-py3-none-any.whl (29 kB)
Using cached platformdirs-3.10.0-py3-none-any.whl (17 kB)
Building wheels for collected packages: shellcheck-py
  Building wheel for shellcheck-py (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for shellcheck-py (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [18 lines of output]
      /private/var/folders/q8/2kspgxtx1fnb2cqscf_nvgph0000gn/T/pip-build-env-qrkd_b_0/overlay/lib/python3.10/site-packages/setuptools/config/setupcfg.py:293: _DeprecatedConfig: Deprecated config in `setup.cfg`
      !!

              ********************************************************************************
              The license_file parameter is deprecated, use license_files instead.

              By 2023-Oct-30, you need to update your project and remove deprecated calls
              or your builds will no longer be supported.

              See https://setuptools.pypa.io/en/latest/userguide/declarative_config.html for details.
              ********************************************************************************

      !!
        parsed = self.parsers.get(option_name, lambda x: x)(value)
      running bdist_wheel
      running build
      running fetch_binaries
      error: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)>
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for shellcheck-py
Failed to build shellcheck-py
ERROR: Could not build wheels for shellcheck-py, which is required to install pyproject.toml-based projects
```

**Comment by asottile** [comment]
you've got some sort of work proxy blocking the download from github

**Comment by neilyio** [comment]
I'd like to mention that installation has been an issue for me too, and haven't figured out how to solve it yet. I was using an adblocker, and saw the same error as @jbnjohnathan. I turned off the blocker and now I'm seeing this:
```
Collecting shellcheck-py
  Using cached shellcheck_py-0.9.0.6.tar.gz (4.2 kB)
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: shellcheck-py
  Building wheel for shellcheck-py (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [50 lines of output]
      running bdist_wheel
      running build
      running fetch_binaries
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/private/var/folders/sz/466g4f5x7j1d5_9wnd5rzsf40000gn/T/pip-install-wnh9ioqh/shellcheck-py_63e55a0e7f20493680e4e2347eb31372/setup.py", line 177, in <module>
          setup(version=f'{SHELLCHECK_VERSION}.{PY_VERSION}', cmdclass=command_overrides)
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/__init__.py", line 103, in setup
          return distutils.core.setup(**attrs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
                 ^^^^^^^^^^^^^^^^^^
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/dist.py", line 989, in run_command
          super().run_command(command)
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/opt/homebrew/lib/python3.11/site-packages/wheel/bdist_wheel.py", line 364, in run
          self.run_command("build")
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/dist.py", line 989, in run_command
          super().run_command(command)
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/command/build.py", line 131, in run
          self.run_command(cmd_name)
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/dist.py", line 989, in run_command
          super().run_command(command)
        File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/private/var/folders/sz/466g4f5x7j1d5_9wnd5rzsf40000gn/T/pip-install-wnh9ioqh/shellcheck-py_63e55a0e7f20493680e4e2347eb31372/setup.py", line 124, in run
          archive = download(url, sha256)
                    ^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/sz/466g4f5x7j1d5_9wnd5rzsf40000gn/T/pip-install-wnh9ioqh/shellcheck-py_63e55a0e7f20493680e4e2347eb31372/setup.py", line 64, in download
          data = resp.read()
                 ^^^^^^^^^^^
        File "/opt/homebrew/Cellar/python@3.11/3.11.6/Frameworks/Python.framework/Versions/3.11/lib/python3.11/http/client.py", line 482, in read
          s = self._safe_read(self.length)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/opt/homebrew/Cellar/python@3.11/3.11.6/Frameworks/Python.framework/Versions/3.11/lib/python3.11/http/client.py", line 633, in _safe_read
          raise IncompleteRead(data, amt-len(data))
      http.client.IncompleteRead: IncompleteRead(3644608 bytes read, 560196 more expected)
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for shellcheck-py
  Running setup.py clean for shellcheck-py
Failed to build shellcheck-py
ERROR: Could not build wheels for shellcheck-py, which is required to install pyproject.toml-based projects

[notice] A new release of pip is available: 23.2.1 -> 23.3.1
[notice] To update, run: python3.11 -m pip install --upgrade pip
```

I haven't had any issues installing any other Python package on this system, either due to the adblocker or any other reason.

**Comment by asottile** [comment]
this is a you-network-problem -- the connection was severed during download.  you need to either retry, be more patient, or debug why your network is low quality
